### PR TITLE
boards: arm: nrf53: add missing Dev Kit information

### PR DESCRIPTION
**Pull request by ioannisg**
Adding missing information (image and link) for the
nRF5340 Development Kit.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

Information was missing in #20157 intentionally, since the SoC was not public (now it is).

### CODE REVIEW

File: boards/arm/nrf5340_dk_nrf5340/doc/index.rst [hunk 1]
```diff
@@ -257,4 +266,6 @@ References
 
 .. _IDAU:
    https://developer.arm.com/products/architecture/cpu-architecture/m-profile/docs/100690/latest/attribution-units-sau-and-idau
+.. _nRF53 DK website:
+   https://www.nordicsemi.com/Software-and-tools/Development-Kits/nRF5340-PDK
 .. _Nordic Semiconductor Documentation library: https://www.nordicsemi.com/DocLib
```
Comment (sigurdnev) [inline boards/arm/nrf5340_dk_nrf5340/doc/index.rst]: DocLib is not longer being used.  We should link to Infocenter instead.

Comment (ioannisg) [reply boards/arm/nrf5340_dk_nrf5340/doc/index.rst]: See PR #20765

**Review by carlescufi — APPROVED** [review]

**Review by dbkinder — APPROVED** [review]
LGTM, thanks.

**Comment by ioannisg** [comment]
I suppose this one can be merged, @aescolar @carlescufi

**Review by ru-fu — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**